controllers/controllers_suite_test.go

```diff
@@ -371,3 +433,137 @@ func setupMachineCRDs() {
 		return ph.Patch(ctx, dummies.CSMachine1, patch.WithStatusObservedGeneration{})
 	}, timeout).Should(Succeed())
 }
+
+// setupMachineCRDs creates a CAPI and CloudStack machine with an appropriate ownership ref between them.
+func setupCAPIMachineAndCSMachineCRDs(CSMachine *infrav1.CloudStackMachine, CAPIMachine *clusterv1.Machine) {
+	//  Create them.
+	Ω(k8sClient.Create(ctx, CAPIMachine)).Should(Succeed())
+	Ω(k8sClient.Create(ctx, CSMachine)).Should(Succeed())
+
+	// Fetch the CS Machine that was created.
+	key := client.ObjectKey{Namespace: dummies.CSCluster.Namespace, Name: CSMachine.Name}
+	Eventually(func() error {
+		return k8sClient.Get(ctx, key, CSMachine)
+	}, timeout).Should(BeNil())
+
+	// Set owner ref from CAPI machine to CS machine and patch back the CS machine.
+	Eventually(func() error {
+		ph, err := patch.NewHelper(dummies.CSMachine1, k8sClient)
+		Ω(err).ShouldNot(HaveOccurred())
+		dummies.CSMachine1.OwnerReferences = append(dummies.CSMachine1.OwnerReferences, metav1.OwnerReference{
+			Kind:       "Machine",
+			APIVersion: clusterv1.GroupVersion.String(),
+			Name:       CAPIMachine.Name,
+			UID:        "uniqueness",
+		})
+		return ph.Patch(ctx, CSMachine, patch.WithStatusObservedGeneration{})
+	}, timeout).Should(Succeed())
+}
+
+// setupCAPIMachineDeploymentCRDs creates a CAPI machine deployment.
+func setupCAPIMachineDeploymentCRDs(CAPIMachineDeployment *clusterv1.MachineDeployment) {
+	//  Create them.
+	Ω(k8sClient.Create(ctx, CAPIMachineDeployment)).Should(Succeed())
+
+	// Fetch the CAPI Machine Deployment that was created.
+	key := client.ObjectKey{Namespace: dummies.ClusterNameSpace, Name: CAPIMachineDeployment.Name}
+	Eventually(func() error {
+		return k8sClient.Get(ctx, key, CAPIMachineDeployment)
+	}, timeout).Should(BeNil())
+}
+
+// setupEtcdadmClusterCRDs create an etcdadm Cluster
+func setupEtcdadmClusterCRDs(etcdadmCluster *etcdadmController.EtcdadmCluster) {
```

Comment:
Since it only creates one CR, the function name should be singular.
```suggestion
func setupEtcdadmClusterCRD(etcdadmCluster *etcdadmController.EtcdadmCluster) {
```